### PR TITLE
Assert that locations are not empty

### PR DESCRIPTION
**Pull request by konstin**
Before that an empty locations list would raise some weird pyahocorasick error about a value not being an automaton

### CODE REVIEW

File: geoextract/__init__.py [hunk 1]
```diff
@@ -188,6 +188,10 @@ def setup(self, pipeline):
 
     def extract(self, text):
         text = self._pad(text)
+
+        if not list(self._automaton.items()):
+            raise ValueError("You must provide locations for extraction")
+
```
Comment (torfsen) [inline geoextract/__init__.py]: I'm fine with adding this check, but it should be done at the end of `NameExtractor.setup` instead. At that point, no further words will be added to the automaton.

**Review by torfsen — CHANGES_REQUESTED** [review]
Thanks for your contribution! There's a small change that I'd like to see. In addition, please add a corresponding test case to the `TestNameExtractor` class.

**Comment by torfsen** [comment]
In hindsight, I think the better approach is to simply return no matches from `NameExtractor.extract` in case the locations list was empty. This is consistent with the expected behavior, doesn't raise any exceptions and doesn't require special code in tests, etc. to avoid an empty locations list.

I've made this change in 0d184e264c8356d22580076d35f16866c8ce6ecc.

**Comment by konstin** [comment]
Ah great, that makes it even easier to use!

**Pull request merged**